js/piece.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Abstract Piece class
+ */
+class Piece {
```

Comment:
Make sure that the class is exported by using the `export` keyword before the `class` keyword so that it can be used in other modules.